docs/site/migration/models/methods.md

```diff
@@ -6,8 +6,309 @@ sidebar: lb4_sidebar
+- [Via server boot script](https://loopback.io/doc/en/lb3/Customizing-models.html#via-server-boot-script)
+- [Via model's script](https://loopback.io/doc/en/lb3/Customizing-models.html#via-your-models-script)
+
+
+The following LB3 code snippets show how to reimplement Note.find() to override the built-in [find()](https://apidocs.loopback.io/loopback/#persistedmodel-find) method.
+
+
+**/server/boot/script.js**
+```js
+module.exports = function(app) {
+  var Note = app.models.Note;
+  var find = Note.find;
+  var cache = {};
+
+  Note.find = function(filter, cb) {
+    var key = '';
+    if(filter) {
+      key = JSON.stringify(filter);
+    }
+    var cachedResults = cache[key];
+    if(cachedResults) {
+      console.log('serving from cache');
+      process.nextTick(function() {
+        cb(null, cachedResults);
+      });
+    } else {
+      console.log('serving from db');
+      find.call(Note, function(err, results) {
+        if(!err) {
+          cache[key] = results;
+        }
+        cb(err, results);
+      });
+    }
+  }
+}
+```
+
+OR
+
+**common/models/Note.js**
+```js
+module.exports = function(Note) {
+  Note.on('dataSourceAttached', function(obj){
+    var find = Note.find;
+    Note.find = function(filter, cb) {
+      return find.apply(this, arguments);
+    };
+  });
+};
+```
+
+In LoopBack 4, data-access APIs are implemented by [repositories](../../Repositories.md) that are decoupled from models.
+
+A Repository represents a specialized Service interface that provides strong-typed data access (for example, CRUD) operations of a domain model against the underlying database or service. A single model can be used with multiple different Repositories. A Repository can be defined and implemented by application developers. LoopBack ships a few predefined Repository interfaces for typical CRUD and KV operations. These Repository implementations leverage Model definition and DataSource configuration to fulfill the logic for data access. See more examples at [Repository/CrudRepository/EntityRepository](https://github.com/strongloop/loopback-next/blob/master/packages/repository/src/repositories/repository.ts) and [KeyValueRepository](https://github.com/strongloop/loopback-next/blob/master/packages/repository/src/repositories/kv.repository.ts).
+
+A CLI generated repository for a model `Note` would look like this:
+
+```ts
+export class NoteRepository extends DefaultCrudRepository<
+  Note,
+  typeof Note.prototype.id,
+  NoteRelations
+> {
+  constructor(
+    @inject('datasources.db') dataSource: DbDataSource,
+  ) {
+    super(Note, dataSource);
+  }
+}
+```
+
+It extends [DefaultCrudRepository](https://github.com/strongloop/loopback-next/blob/master/packages/repository/src/repositories/legacy-juggler-bridge.ts). To customize the `find` method, you would override this method in the `NoteRepository` class.
+
```

Comment:
Can you please add a code snippet showing how a custom version of `find` would look like? Ideally, the snippet should show the outcome of migrating the custom method described earlier in LB3 code snippet.